vlog4j-parser/src/test/java/org/semanticweb/vlog4j/syntax/parser/RuleParserTest.java

```diff
@@ -412,6 +387,27 @@ public void testBlankPredicateName() throws ParsingException {
 		RuleParser.parse(input);
 	}
 
+	@Test(expected = ParsingException.class)
+	public void parseLiteral_invalidLiteralString_throws() throws ParsingException {
+		final String input = "P(\"a\")^^whatever";
+		RuleParser.parseLiteral(input);
+	}
+
+	@Test(expected = ParsingException.class)
+	public void testNonIriTypeInDatatypeLiteral() throws ParsingException {
+		final String input = "\"a\"^^whatever";
+		RuleParser.parseTerm(input);
+	}
+
+	@Test
+	public void testIriTypeInDatatypeLiteral() throws ParsingException {
+		final String iri = "whatever";
+		final String input = "P(\"a\"^^<" + iri + ">)";
+		Literal literal = RuleParser.parseLiteral(input);
+		DatatypeConstant result = (DatatypeConstant) literal.getConstants().toArray()[0];
+		assertEquals(iri, result.getDatatype());
+	}
+
 	@Test
 	public void predicateRelativeNumericIRITest() throws ParsingException {
```

Comment:
the IRI does not seem numeric

Reply:
`1.e1` is a valid `XSD_DOUBLE` literal, though.